front/src/applications/stdcm/types.ts

```diff
@@ -54,6 +55,35 @@ export type StdcmPathProperties = {
   zones: NonNullable<PathProperties['zones']>;
 };
 
+export type RefSchedule = {
+  begin: SimilarTrainWaypointResponse;
+  end: SimilarTrainWaypointResponse;
+  start_time?: string | null;
+  train_id: string;
+};
+
+type WaypointWithSecondaryCode = SimilarTrainWaypointResponse & {
+  secondary_code?: string | null;
+};
+
+export type RefScheduleWithSecondaryCode = {
+  train_id: string;
+  start_time?: string | null;
```

Comment:
Can you convert the start_time in Date here ? And why can it be null (maybe we should discuss this with the backend :thinking:) ?

Reply:
it can be null when there is no similar train found, the back still returns an object but with this value to null

Reply:
What I meant was that you could transform the `string | undefined | null` type => `undefined | Date`
```suggestion
  start_time?: Date;
```

Reply:
but the back returns 

```js
export type PostSimilarTrainsApiResponse =
  /** status 200 A combination of reference train identifiers similar to the provided train */ {
    similar_trains: {
      begin: SimilarTrainWaypointResponse;
      end: SimilarTrainWaypointResponse;
      /** Both `train_name` and `start_time` are `None` if no similar train
        was found for the segment; otherwise, both are `Some`. */
      start_time: string | null;
      /** Both `train_name` and `start_time` are `None` if no similar train
        was found for the segment; otherwise, both are `Some`. */
      train_name: string | null;
    }[];
  };
  ```

  so yes undefined is not required but can't change the rest for the moment

Reply:
done